x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mappings_logic.ts

```diff
@@ -0,0 +1,337 @@
+          if (value.length === 1) return value;
+          if (!newSelectionsContainAny) return value;
+          if (previouslyContainedAny) return value.filter((v) => v !== ANY_AUTH_PROVIDER);
+          return [ANY_AUTH_PROVIDER];
+        },
+        setRoleMappingData: (_, { roleMapping }) =>
+          roleMapping ? roleMapping.authProvider : [ANY_AUTH_PROVIDER],
+      },
+    ],
+  },
+  listeners: ({ actions, values }) => ({
+    initializeRoleMappings: async () => {
+      const { http } = HttpLogic.values;
+      const route = '/api/app_search/role_mappings';
+
+      try {
+        const response = await http.get(route);
+        actions.setRoleMappingsData(response);
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+    },
+    initializeRoleMapping: async ({ roleId }) => {
+      const { http } = HttpLogic.values;
+      const { navigateToUrl } = KibanaLogic.values;
+      const route = roleId
+        ? `/api/app_search/role_mappings/${roleId}`
+        : '/api/app_search/role_mappings/new';
+
+      try {
+        const response = await http.get(route);
+        actions.setRoleMappingData(response);
+      } catch (e) {
+        navigateToUrl(ROLE_MAPPINGS_PATH);
+        flashAPIErrors(e);
+      }
+    },
+    handleDeleteMapping: async () => {
+      const { roleMapping } = values;
+      if (!roleMapping) return;
+
+      const { http } = HttpLogic.values;
+      const { navigateToUrl } = KibanaLogic.values;
+      const route = `/api/app_search/role_mappings/${roleMapping.id}`;
+
+      if (window.confirm(DELETE_ROLE_MAPPINGS_MESSAGE)) {
+        try {
+          await http.delete(route);
+          navigateToUrl(ROLE_MAPPINGS_PATH);
+        } catch (e) {
+          flashAPIErrors(e);
+        }
+      }
+    },
+    handleResetMappings: async (callback) => {
+      const { http } = HttpLogic.values;
+      try {
+        await http.post('/api/app_search/role_mappings/reset');
+        actions.initializeRoleMappings();
+      } catch (e) {
+        flashAPIErrors(e);
+      } finally {
+        callback();
+      }
+    },
+    handleSaveMapping: async () => {
+      const { http } = HttpLogic.values;
+      const { navigateToUrl } = KibanaLogic.values;
+
+      const {
+        attributeName,
+        attributeValue,
+        roleType,
+        roleMapping,
+        accessAllEngines,
+        selectedEngines,
+        selectedAuthProviders: authProvider,
+      } = values;
+
+      const body = JSON.stringify({
+        roleType,
+        accessAllEngines,
+        authProvider,
+        rules: {
+          [attributeName]: attributeValue,
+        },
+        engines: accessAllEngines ? [] : Array.from(selectedEngines),
+      });
+
+      const request = !roleMapping
+        ? http.post('/api/app_search/role_mappings', { body })
+        : http.put(`/api/app_search/role_mappings/${roleMapping.id}`, { body });
+
+      try {
+        await request;
+        navigateToUrl(ROLE_MAPPINGS_PATH);
```

Comment:
similar-ish question, do we want a queued success message before navigation?

Reply:
Agreed. Nice catch!

Reply:
Added here (https://github.com/elastic/kibana/pull/94461/commits/208029612c565e67114f9b0cfde0404267d3ee3b). I could not, for the life of me, get `setQueuedSuccessMessages` to work before the `navigateToUrl` call. Doing it this way worked fine.